### DISCUSSION TITLE
`x-ref` doesn't set `$refs`

### DISCUSSION THREAD

**Discussion by notramo** [alpinejs/alpine#3383] [7. Bugs (archived)]
The `x-ref` example from the alpinejs.dev site doesn't work.
This is a minimal HTML to reproduce.
```html
<!DOCTYPE html>
<html>
  <meta charset="utf-8"/>
  <meta name="viewport" content="width=device-width,initial-scale=1"/>
  <script src="https://unpkg.com/alpinejs" defer></script>
  <title>Bug report</title>
  <style>
    [x-cloak] {
      display: none;
    }
  </style>
</html>
<body>
  <!-- x-ref example from alpinejs.dev -->
  <input type="text" x-ref="content">

   <button x-init x-on:click="navigator.clipboard.writeText($refs.content.value)">
       Copy
   </button>
   <!-- end example -->

   <button x-init x-on:click="console.log($refs.content)">
       Log value
   </button>
</body>
```

**Comment by SimoTod** [comment]
You don't have x-data (you also don't need all those x-init. Yeah, I know you added them because the click wasn't working but again the problem is that you don't have a common ancestor with x-data).

**Reply by notramo** [reply]
Thanks. The documentation didn't mention it.

**Reply by ekwoka** [reply]
https://alpinejs.dev/directives/data

**Reply by flopacero** [reply]
@ekwoka I don't see it mentions `x-data` is required for `x-ref` to work. And there's nothing in `x-ref` docs

**Reply by ekwoka** [reply]
Everything requires x-data. (Except x-init)
